packages/core/src/hooks.ts

```diff
@@ -51,7 +51,7 @@ export function initializeConnector<T extends Connector>(
 }
 
 function computeIsActive({ chainId, accounts, activating }: Web3ReactState) {
-  return Boolean(chainId && accounts && !activating)
+  return chainId && accounts?.length && !activating
```

Comment:
This should still be cast to a boolean (using `Boolean(...)`) so that `undefined | 0` is not returned.